### DISCUSSION TITLE
execute RedirectToCanonicalUrl before UseErrorPages

### DISCUSSION THREAD

**Discussion by parmarnishant** [Dotnet-Boxed/Templates#130] [Q&A]
Hi,
I am using both RedirectToCanonicalUrl and UseStatusCodePagesWithReExecute in cms project.
My issue is that 404 Error Page is displayed before redirecting to Canonical Url.
For example, if "site/pagename" is valid url
"site/Pagename" should first redirect to "site/pagename" , and then if not found display 404 page.

**Comment by RehanSaeed** [comment]
Will investigate.

**Comment by RehanSaeed** [comment]
The templates behaviour is that if the page is not found there is no redirect. If you have changed the middleware pipeline order, this may have caused what you are seeing. Also the 404 page is only visible using the Production environment.
